src/core/bluetoothreceiver.cpp

```diff
@@ -19,8 +19,12 @@
 #include <QDebug>
 #include <QSettings>
 
-BluetoothReceiver::BluetoothReceiver( QObject *parent )
-  : QObject( parent ), mLocalDevice( std::make_unique<QBluetoothLocalDevice>() ), mSocket( new QBluetoothSocket( QBluetoothServiceInfo::RfcommProtocol ) ), mGpsConnection( std::make_unique<QgsNmeaConnection>( mSocket ) )
+BluetoothReceiver::BluetoothReceiver( const QString &address, QObject *parent )
+  : AbstractGnssReceiver( parent )
+  , mAddress( address )
+  , mLocalDevice( std::make_unique<QBluetoothLocalDevice>() )
+  , mSocket( new QBluetoothSocket( QBluetoothServiceInfo::RfcommProtocol ) )
```

Comment:
I was looking if we could get rid of `new` altogether or mostly for less memory headaches ...


```cpp
template <class T, typename... Args>
auto q_make_unique(Args&&... args)
{
auto tmp = std::make_unique<T>(std::forward<Args>(args)...);
auto deleter = [q_ptr = QPointer<T>(tmp.get())](const T* obj)
{
if (q_ptr) {
delete obj;
}
};
```

Ehat about introducing `q_make_unique`?

Reply:
I'm +1 on the idea, but I'd rather us work on this in a separate PR that's not already 1,200 lines of changes :) 

_(This specific use of `new` was already there beforehand so we're not making things worse on that front.)_